src/main/lrsql/spec/admin/status.clj

```diff
@@ -0,0 +1,114 @@
+(ns lrsql.spec.admin.status
+  (:require [clojure.spec.alpha :as s]
+            [lrsql.backend.protocol :as bp]
+            [lrsql.spec.common :as common]
+            [lrsql.util :as u]
+            [xapi-schema.spec :as xs]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Interface
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn admin-status-backend?
+  [bk]
+  (satisfies? bp/AdminStatusBackend bk))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Axioms
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(s/def ::count nat-int?)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Inputs
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(s/def ::status-query-name
+  #{"statement-count"
+    "actor-count"
+    "platform-frequency"
+    "last-statement-stored"
+    "timeline"})
+
+(s/def :lrsql.spec.admin.status.params/include
+  (s/nonconforming
+   (s/or :single   ::status-query-name
+         :multiple (s/every ::status-query-name))))
+(s/def :lrsql.spec.admin.status.params/timeline-unit
+  #{"year"
+    "month"
+    "day"
+    "hour"
+    "minute"
+    "second"})
+
+(defn- after-epoch?
+  [timestamp]
+  (-> timestamp
+      u/str->time
+      u/time->millis
+      nat-int?))
+
+(s/def :lrsql.spec.admin.status.params/timeline-since
+  (s/and ::xs/timestamp
+         after-epoch?))
+(s/def :lrsql.spec.admin.status.params/timeline-until
+  (s/and ::xs/timestamp
+         after-epoch?))
+
+
```

Comment:
Extra space? (Also there's some wonky spacing in the rest of this namespace)

Reply:
Do you mean extra line break? Sure I can remove it.

Reply:
https://github.com/yetanalytics/lrsql/pull/282/commits/be2d844fa26c37d66531a41901d1e8494cd03605